cmd/svcat/class/get_cmd_test.go

```diff
@@ -174,3 +177,126 @@ func TestListClasses(t *testing.T) {
+			fakeApp.SvcatClient = fakeSDK
+			cmd := getCmd{
+				Namespaced: &command.Namespaced{Context: svcattest.NewContext(outputBuffer, fakeApp)},
+				Scoped:     command.NewScoped(),
+			}
+			cmd.Namespace = "default"
+			cmd.Scope = servicecatalog.NamespaceScope
+
+			err := cmd.getAll()
+
+			Expect(err).NotTo(HaveOccurred())
+			scopeArg := fakeSDK.RetrieveClassesArgsForCall(0)
+			Expect(scopeArg).To(Equal(servicecatalog.ScopeOptions{
+				Namespace: "default",
+				Scope:     servicecatalog.NamespaceScope,
+			}))
+
+			output := outputBuffer.String()
+			Expect(output).To(ContainSubstring("mysqldb"))
+		})
+		It("Calls the pkg/svcat libs RetrieveClasses with namespace scope and all namespaces", func() {
+			outputBuffer := &bytes.Buffer{}
+
+			fakeApp, _ := svcat.NewApp(nil, nil, "default")
+			fakeSDK := new(servicecatalogfakes.FakeSvcatClient)
+			fakeSDK.RetrieveClassesReturns(
+				[]servicecatalog.Class{
+					&v1beta1.ServiceClass{ObjectMeta: metav1.ObjectMeta{Name: "mysqldb", Namespace: "default"}},
+					&v1beta1.ServiceClass{ObjectMeta: metav1.ObjectMeta{Name: "postgresdb", Namespace: "test-ns"}},
+				},
+				nil)
+			fakeApp.SvcatClient = fakeSDK
+			cmd := getCmd{
+				Namespaced: &command.Namespaced{Context: svcattest.NewContext(outputBuffer, fakeApp)},
+				Scoped:     command.NewScoped(),
+			}
+			cmd.Namespace = ""
+			cmd.Scope = servicecatalog.NamespaceScope
+
+			err := cmd.getAll()
+
+			Expect(err).NotTo(HaveOccurred())
+			scopeArg := fakeSDK.RetrieveClassesArgsForCall(0)
+			Expect(scopeArg).To(Equal(servicecatalog.ScopeOptions{
+				Namespace: "",
+				Scope:     servicecatalog.NamespaceScope,
+			}))
+
+			output := outputBuffer.String()
+			Expect(output).To(ContainSubstring("mysqldb"))
+			Expect(output).To(ContainSubstring("postgresdb"))
+		})
+		It("Calls the pkg/svcat libs RetrieveClasses with all scope and current namespaces", func() {
+			outputBuffer := &bytes.Buffer{}
+
+			fakeApp, _ := svcat.NewApp(nil, nil, "default")
+			fakeSDK := new(servicecatalogfakes.FakeSvcatClient)
+			fakeSDK.RetrieveClassesReturns(
+				[]servicecatalog.Class{
+					&v1beta1.ClusterServiceClass{ObjectMeta: metav1.ObjectMeta{Name: "mysqldb"}},
+					&v1beta1.ServiceClass{ObjectMeta: metav1.ObjectMeta{Name: "postgresdb", Namespace: "default"}},
+				},
+				nil)
+			fakeApp.SvcatClient = fakeSDK
+			cmd := getCmd{
+				Namespaced: &command.Namespaced{Context: svcattest.NewContext(outputBuffer, fakeApp)},
+				Scoped:     command.NewScoped(),
+			}
+			cmd.Namespace = "default"
+			cmd.Scope = servicecatalog.AllScope
+
+			err := cmd.getAll()
+
+			Expect(err).NotTo(HaveOccurred())
+			scopeArg := fakeSDK.RetrieveClassesArgsForCall(0)
+			Expect(scopeArg).To(Equal(servicecatalog.ScopeOptions{
+				Namespace: "default",
+				Scope:     servicecatalog.AllScope,
+			}))
+
+			output := outputBuffer.String()
+			Expect(output).To(ContainSubstring("mysqldb"))
+			Expect(output).To(ContainSubstring("postgresdb"))
+		})
+	})
+})
```

Comment:
Would also be nice to have a set of cases for when you're just getting  a single class

Reply:
👍